### ISSUE TITLE
Add an API for Webhook Management

### ISSUE THREAD

**Issue by mitsuhiko** [getsentry/sentry#3482]
After we sent emails to people about rotating webhooks it came in as feature request to be able to manage webhooks from the API.

**Comment by atmos** [comment]
Did anything ever come out of this? I can't seem to find a way to programmatically add/update webhooks across my organization's usage.

**Comment by dcramer** [comment]
@atmos not yet -- though theres another (unrelated branch) where we're moving plugin config into undocumented APIs:

https://github.com/getsentry/sentry/pull/3947

It's likely we're going to pull in some of the changes as they're unrelated to that specific branch, and that branch is turning into a pretty hefty change.

**Comment by atmos** [comment]
So for the time being I should have my teams enable webhooks manually? I'm moving 100+ projects over and was trying to avoid having everyone copy and paste things, it's also a pita if I need to roll creds.

**Comment by dcramer** [comment]
#3986

**Comment by dcramer** [comment]
@atmos is this on-premise install? hosted? can probably help one way or the other if theres a way to export them from the current system

**Comment by atmos** [comment]
@dcramer Yeah, it's on-premise. I have the ability to get all the projects from the other system, and it looks like I can create them via the API. My main concern is updating all of this information in the event of a credential roll. Ideally I'd just have organization or team-wide webhooks.

**Comment by dcramer** [comment]
@atmos tbh might make sense to throw in a small python plugin that just sends webhooks itself (and can be enabled by default):

https://github.com/getsentry/sentry/blob/master/src/sentry/plugins/sentry_webhooks/plugin.py

Then just define:

```
project_default_enabled = True
```

And ensure it gets registered somehow when process loads:

https://github.com/getsentry/sentry/blob/master/src/sentry/plugins/sentry_webhooks/__init__.py

(Easiest is to stuff it as a loadable into INSTALLED_APPS in sentry.conf.py)

**Comment by atmos** [comment]
Can I distribute it as with pip or do I need to fork the project?

**Comment by mattrobenolt** [comment]
Plugins are pretty simple to be done by themselves. You basically just need to register the required `entry_points` and they'll be loaded automatically after installed. See something like https://github.com/getsentry/sentry-github/blob/master/setup.py#L39-L46 for an example.

**Comment by dcramer** [comment]
@atmos yeah you could just fork something like our sentry-slack repo to use as a base then pip install <url/path to zip file>. Main thing is it needs the setup.py to be configured correctly, and thats why I'd suggest using one of our repos as an example.

Also if you do it that way you dont need to worry about INSTALLED_APPS as we do some Python magic.

Only downside is we document basically none of the plugin development

tl;dr take sentry-slack, replace plugin with sentry_webhook's plugin.py, update setup.py, modify to your heart's content, and pip install the .zip

**Comment by atmos** [comment]
I consolidated the two and found two different method signatures for notifying. Which one is more applicable?

```python
    def notify_users(self, group, event, fail_silently=False):
        payload  = self.get_group_data(group, event)
        endpoint = getattr(settings, 'SENTRY_ORG_WEBHOOK_URI', 3)
        safe_execute(self.send_webhook, endpoint, payload, _with_transaction=False)

    def notify(self, notification):
```

**Comment by dcramer** [comment]
@atmos you probably just want notify:

https://github.com/getsentry/sentry/blob/master/src/sentry/plugins/bases/notify.py#L58

**Comment by dcramer** [comment]
Closing this out as this was landed with parts of our plugin refactor.

```
/api/0/projects/:org/:project/plugins/:plugin/
```

**Comment by atmos** [comment]
So if we update to 8.8.0 we should be all set to use this?

**Comment by dcramer** [comment]
@atmos yeah 8.8.0 has the basics of it, but 8.9.0 will likely include some changes and additions as we rough out more of the new systems (I'm not yet sure if they'll be backwards compatible).

**Comment by Jefreesujit** [comment]
could anyone here guide me how to configure web hooks in sentry? I ve tried adding the callback url but the endpoint is not hitting..

**Comment by eduardomendonca-smn** [comment]
@Jefreesujit Did you get something about your question? 

I have a very old application and I would like to use the API instead of the SDK that is not supported and my application is in .Net 2.0

[(https://docs.sentry.io/api/)]
@dcramer  Is this Sentry API the same as the SDK? Can I use it to trace events I code my webforms in C #?
